src/datepicker/datepicker-service.ts

```diff
@@ -78,7 +78,7 @@ export class NgbDatepickerService {
   }
 
   open(date: NgbDate) {
-    if (!this._state.disabled && this._calendar.isValid(date)) {
+    if (this._calendar.isValid(date)) {
```

Comment:
I don't think it should be possible to do anything with the datepicker when it is disabled, so I wouldn't change that. Plus for this PR it is not necessary.

Reply:
If we don't change this, it won't work if the user actually want to see the disabled date picker popup, because the popup won't open at all.

Reply:
I've checked out the code in the PR and it works perfectly fine without this change

`open()` is the `NgbDatepickerService` method that has nothing to do with the popup

The popup works because when you open it, you first

```javascript
// 1. focus()
this._cRef.instance.focus();

// 2. set disabled state
this._setDatepickerDiabledState(this.disabled);
```

it would not work if you set disabled state before calling focus

Reply:
yeah it makes sense, I think it wasn't working before because I set disabled before focus